lib/ansible/modules/cloud/amazon/kinesis_stream.py

```diff
@@ -804,6 +858,36 @@ def update(client, current_stream, stream_name, retention_period=None,
             )
             return success, changed, err_msg
 
+    if current_stream['OpenShardsCount'] != number_of_shards:
+        success, err_msg = (
+            update_shard_count(client, stream_name, number_of_shards, check_mode=check_mode)
+        )
+
+        if not success:
+            return success, changed, err_msg
+
+        changed = True
```

Comment:
Sorry I'm not entirely sure what this comment is about. At this point the update operation has been executed and returns no error. Therefore, marking the status to changed seems correct to me?